### PR TITLE
Container: optimized getByType() performance

### PR DESCRIPTION
**Pull request by JanTvrdik**

### CODE REVIEW

File: src/DI/Container.php [hunk 1]
```diff
@@ -98,8 +98,10 @@ public function removeService($name)
 	 */
 	public function getService($name)
 	{
-		$name = isset($this->meta[self::ALIASES][$name]) ? $this->meta[self::ALIASES][$name] : $name;
 		if (!isset($this->registry[$name])) {
+			if (isset($this->meta[self::ALIASES][$name])) {
+				return $this->getService($this->meta[self::ALIASES][$name]);
+			}
```
Comment (dg) [inline src/DI/Container.php]: Do you think that calling method is faster than accessing array?

Comment (JanTvrdik) [inline src/DI/Container.php]: No

**Comment by JanTvrdik** [comment]
What about this?

File: src/DI/Container.php (hunk 1 above)
Comment (JanTvrdik) [inline src/DI/Container.php]: The method call could be inlined, do you like it better?

~~~php
public function getService($name)
{
	if (!isset($this->registry[$name])) {
		if (isset($this->meta[self::ALIASES][$name])) {
			$name = $this->meta[self::ALIASES][$name];
			if (!isset($this->registry[$name])) {
				$this->registry[$name] = $this->createService($name);
			}
		} else {
			$this->registry[$name] = $this->createService($name);
		}
	}
	return $this->registry[$name];
}
~~~

Comment (dg) [inline src/DI/Container.php]: It is much more complicated. Did you measure benefit in realworld usage?

Comment (JanTvrdik) [inline src/DI/Container.php]: That I prefer and propose the version if extra `getService` call. It is a simple change which improve `getService` performance in most-common use-case (no aliases).

Comment (dg) [inline src/DI/Container.php]: One lookup in aliases is IMHO immeasurable.

Comment (JanTvrdik) [inline src/DI/Container.php]: But why do it, when you can choose not to do it and still keep the code simple?

Comment (dg) [inline src/DI/Container.php]: You are right that aliases are used very rare.

**Pull request merged**